pyodata/v2/model.py

```diff
@@ -1048,6 +1050,8 @@ def association_set(self, set_name, namespace=None):
             except KeyError:
                 pass
 
+        raise KeyError('Association set does not exist')
+
```

Comment:
This could be nice, separate PR, immediately merged, unrelated to the new observer class.